docs/references/kontena-yml.md

```diff
@@ -419,25 +424,29 @@ user: app_user
 
 Mount paths as volumes, optionally specifying a path on the host machine. (HOST:CONTAINER), or an access mode (HOST:CONTAINER:ro).
 
-Data volume:
+##### Named volume (must be defined in the top-level volumes key):
 
 ```
 volumes:
- - /var/lib/mysql
+  - mysql-data:/var/lib/mysql
 ```
 
-Bind mount host directory as a volume:
+Named volumes must be defined in the top-level [volumes](#volumes-configuration-reference) key.
+
+##### Data volume:
 
 ```
 volumes:
- - /data/mysql:/var/lib/mysql
+  - /var/lib/mysql
 ```
 
-Named volumes are supported in the service declaration, but not in the top-level `volumes` key. When defining named volume in the service declaration the default driver configured by the Docker Engine will be used (in most cases, this is the local driver). If volume does not exist it will be created.
+Data volume is created on the fly. Volume is persistent only if service is marked as `stateful: true`.
```

Comment:
Anonymous data volumes are created on the fly. The volume is only persistent if the service is marked as `stateful: true`.